### PR TITLE
Adds TypeScript flavor packages for ESLint

### PR DESCRIPTION
**Pull request by andyfleming**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 
-Official GoDaddy JavaScript styleguide. It includes `eslint` packages for three use-cases and can be used as a standard in any new project.
+Official GoDaddy JavaScript styleguide. It includes `eslint` packages for various use-cases and can be used as a standard in any new project.
 
 - [`eslint-config-godaddy`]: Base configuration for _non-React_, ES6 JavaScript applications
 - [`eslint-config-godaddy-react`]: Configuration for ES6 React JavaScript applications
 - [`eslint-config-godaddy-es5`]: Configuration for React _and_ non-React ES5 JavaScript applications
+- [`eslint-config-godaddy-typescript`]: Configuration for ES6 TypeScript applications
+- [`eslint-config-godaddy-react-typescript`]: Configuration for ES6 React JavaScript applications
+- [`eslint-config-godaddy-flow`]: Configuration for ES6 React JavaScript applications using Flow
```
Comment (fabiomcosta) [inline README.md]: Cool!
Thx for adding this missing one

Comment (andyfleming) [reply README.md]: No problem! 🙂

File: packages/eslint-config-godaddy/cli.js [hunk 2]
```diff
@@ -25,8 +25,8 @@ module.exports = function (filename) {
   // Only force our config file if there is one not in the current
   // directory AND not specified by the command line.
   //
-  fs.readdir(cwd, function (err, files) {
-    if (err) { throw err; }
+  fs.readdir(cwd, function (readErr, files) {
+    if (readErr) { throw readErr; }
```
Comment (andyfleming) [inline packages/eslint-config-godaddy/cli.js]: Changed this to pass our own lint.

File: packages/eslint-config-godaddy-react-typescript/package.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    "typescript"
+  ],
+  "bin": {
+    "eslint-godaddy-react-typescript": "bin/eslint-godaddy-react-typescript"
+  },
+  "dependencies": {
+    "eslint-config-godaddy-react": "^5.0.0"
+  },
+  "peerDependencies": {
+    "@typescript-eslint/eslint-plugin": "^2.3.1",
```
Comment (DullReferenceException) [inline packages/eslint-config-godaddy-react-typescript/package.json]: I would avoid this anti-pattern, which has now been fixed in https://github.com/godaddy/javascript/pull/89 for the existing code. Let's have eslint plugins as regular dependencies, not force consuming apps to install these plugins. The only peer dependency should be eslint and probably typescript in our case.

**Comment by fabiomcosta** [comment]
https://github.com/godaddy/javascript/pull/88 has been merged, so I would adjust this PR to have less peerDeps, according to what has been done there.
This looks fantastic!

File: packages/eslint-config-godaddy-typescript/index.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+module.exports = {
+  extends: [
+    require('./extends')('eslint-config-godaddy')
+  ],
+  parser: '@typescript-eslint/parser',
+  plugins: ['@typescript-eslint'],
+  overrides: [
+    {
+      files: ['*.js', '*.ts', '*.tsx'],
```
Comment (DullReferenceException) [inline packages/eslint-config-godaddy-typescript/index.js]: Should include `.json` files. Also, do we need `*.tsx` for this variation?

File: packages/eslint-config-godaddy-react-typescript/index.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+module.exports = {
+  extends: [
+    require('./extends')('eslint-config-godaddy-react'),
+    'plugin:eslint-plugin-react/recommended'
+  ],
+  parser: '@typescript-eslint/parser',
+  plugins: ['@typescript-eslint'],
+  overrides: [
+    {
+      files: ['*.js', '*.ts', '*.tsx'],
```
[the remark this answers is not in the document]
Comment (andyfleming) [reply packages/eslint-config-godaddy-react-typescript/index.js]: Sure, I'll add those. Good call.

File: packages/eslint-config-godaddy-typescript/index.js (hunk 4 above)
Comment (andyfleming) [reply packages/eslint-config-godaddy-typescript/index.js]: Probably not. Similarly I can remove the prop types rule below.

File: packages/eslint-config-godaddy-react-typescript/package.json (hunk 3 above)
Comment (andyfleming) [reply packages/eslint-config-godaddy-react-typescript/package.json]: Take a look at the latest version of this and see if it aligns with what you have in mind.

**Pull request merged**